reader.go

```diff
@@ -41,31 +41,20 @@ func (d *Reader) Scan() bool {
 			}
 			continue
 		}
-		if !strings.HasPrefix(line, ": ") {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
-		}
-		i := strings.IndexRune(line[2:], ':')
-		if i < 0 {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
-		}
-		time, err = strconv.Atoi(line[2 : i+2])
-		if err != nil {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
-		}
-		j := strings.IndexRune(line[2:], ';')
-		if j < 0 {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
+
+		switch {
+		case strings.HasPrefix(line, ": "):
+			time, elapsed, cmd, err = extended(line)
+		default:
+			cmd = line
 		}
-		elapsed, err = strconv.Atoi(line[i+3 : j+2])
+
 		if err != nil {
```

Comment:
Move just after extended().

Reply:
Can you please clarify what you mean here? This has me slightly confused.

Reply:
No need to check err for standard format.